core/src/main/java/org/elasticsearch/common/lucene/all/AllTermQuery.java

```diff
@@ -64,8 +64,9 @@ public AllTermQuery(Term term) {
 
     @Override
     public Query rewrite(IndexReader reader) throws IOException {
-        if (getBoost() != 1f) {
-            return super.rewrite(reader);
+        Query rewritten = super.rewrite(reader);
```

Comment:
Does this change come from the boost change or something else? It feels related but not the same.